### PR TITLE
Corrected incorrect test assertion handling logic

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: bin/newman.js [hunk 1]
```diff
@@ -30,10 +30,12 @@ cli(process.argv.slice(2), 'newman', function (err, args) {
...
     }
-    dispatch(args, function (err) {
-        if (err) {
-            console.error(err);
-            process.exit(_.get(args, 'run.suppressExitCode') ? 0 : 1);
+    dispatch(args, function (err, summary) {
+        var runError = err || summary.run.failures.length || summary.run.error;
+
+        if (runError && !_.get(args, 'run.suppressExitCode')) {
+            console.error(runError);
```
Comment (czardoz) [inline bin/newman.js]: what? it'll print out the number of failures here

File: test/unit/run.test.js [hunk 2]
```diff
@@ -70,4 +71,98 @@ describe('run module', function () {
...
+                    suppressExitCode: true
+                }, function (err) {
+                    expect(err).be(null);
+                    next();
+                }).not.throwException();
+            }
+        ], done);
+    });
+
+    it('exits with a correct exit code for all 4 --bail, -x combinations (devil-collection)', function (done) {
```
Comment (czardoz) [inline test/unit/run.test.js]: These tests do not actually test for exit code, they test for the error object

File: test/unit/run.test.js [hunk 3]
```diff
@@ -70,4 +71,98 @@ describe('run module', function () {
             }
         ], done);
     });
+
+    it('exits with code 0 for all 4 --bail, -x combinations (angel-collection)', function (done) {
```
Comment (czardoz) [inline test/unit/run.test.js]: where's the exit code being tested?

File: bin/newman.js (hunk 1 above)
Comment (shamasis) [inline bin/newman.js]: Remove console.error from here and move it to `done` callback of CLI reporter. talk to @czardoz

File: examples/angel.postman_collection.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+				"body": {
+					"mode": "formdata",
+					"formdata": []
+				},
+				"description": "A request that always fails at the tests"
+			},
+			"response": []
+		}
+	]
+}
```
Comment (shamasis) [inline examples/angel.postman_collection.json]: no eof newline

File: examples/devil.postman_collection.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+				"body": {
+					"mode": "formdata",
+					"formdata": []
+				},
+				"description": "A request that always fails at the tests"
+			},
+			"response": []
+		}
+	]
+}
```
Comment (shamasis) [inline examples/devil.postman_collection.json]: eof newline missing

**Comment by shamasis** [comment]
@kunagpal please remove creatively named collections from examples. someone reading the repository would not understand from file name what the collection is

File: test/unit/run.test.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
-    async = require('async');
+var path = require('path'),
+    async = require('async'),
+    expect = require('expect.js'),
+    sdk = require('postman-collection');
 
 /* global describe, it */
 describe('run module', function () {
-    var run = require('../../lib/run');
+    var run = require(path.join(__dirname, '..', '..', 'lib', 'run'));
```
Comment (shamasis) [inline test/unit/run.test.js]: why? windows is compatible without doing this.

File: test/unit/run.test.js [hunk 7]
```diff
@@ -70,4 +71,114 @@ describe('run module', function () {
             }
         ], done);
     });
+
+    it('exits with code 0 for all 4 --bail, -x combinations (angel-collection)', function (done) {
+        var sampleCollection = path.join(__dirname, '..', '..', 'examples', 'angel.postman_collection.json');
```
Comment (shamasis) [inline test/unit/run.test.js]: Please have inline fixtures for these tests! No referencing stuff.

**Comment by shamasis** [comment]
@czardoz: made the following changes

- removed collections from examples folder since they were not localised to tests
- wrote a cli test running framework
- fixed issue where cli arg parse failures were not returning exit code
- removed unit tests of run.js that had no side effect

**Pull request merged**